OpenRA.Game/GameRules/ActorInfo.cs

```diff
@@ -184,8 +184,13 @@ static IEnumerable<Type> PrerequisitesOf(ITraitInfo info)
 					i.Name.Replace("Init", ""), i));
 		}
 
-		public bool HasTraitInfo<T>() where T : ITraitInfo { return Traits.Contains<T>(); }
-		public T TraitInfo<T>() where T : ITraitInfo { return Traits.Get<T>(); }
-		public T TraitInfoOrDefault<T>() where T : ITraitInfo { return Traits.GetOrDefault<T>(); }
+		public bool HasTraitInfo<T>() where T : ITraitInfo { return traits.Contains<T>(); }
+		public T TraitInfo<T>() where T : ITraitInfo { return traits.Get<T>(); }
+		public T TraitInfoOrDefault<T>() where T : ITraitInfo { return traits.GetOrDefault<T>(); }
+		public IEnumerable<ITraitInfo> TraitInfos() { return traits.WithInterface<ITraitInfo>(); }
+		public IEnumerable<T> TraitInfos<T>() where T : ITraitInfo { return traits.WithInterface<T>(); }
+
+		// For testing only
+		public void AddTraitInfo(ITraitInfo traitInfo) { traits.Add(traitInfo); }
```

Comment:
This is a bit ugly.  Can you make the unit test use the standard mechanism for adding traitinfos, or hack it in that one place using reflection?